MmSupervisorPkg/Core/MmSupervisorCore.c

```diff
@@ -711,23 +712,41 @@ DiscoverStandaloneMmDriversInFvHobs (
   EFI_PEI_HOB_POINTERS            Hob;
   EFI_STATUS                      Status;
 
+  UINT16  HobType;
+
   Hob.Raw = GetHobList ();
   if (Hob.Raw == NULL) {
     return EFI_NOT_FOUND;
   }
 
+  // If PcdMigrateTemporaryRamFirmwareVolumes is set, the
+  // migration of temporary RAM firmware volumes can end up
+  // unmapping the fvs, so use Fv2 hobs.
+  if (FixedPcdGetBool (PcdMigrateTemporaryRamFirmwareVolumes)) {

```

Comment:
This is a platform bug. They should not pass any HOBs that are not mapped regardless of whether firmware volumes are migrated (they should always be on Intel platforms for security and perf reasons).

This PCD value is also in a separate build from the platform for SEA builds where the supervisor core is delivered as a binary so the `FixedAtBuild` value is irrelevant to the platform's setting.